components/media/Feed.tsx

```diff
@@ -68,36 +46,56 @@ const Feed = ({
     }
   }, [fetchNextPage, hasNextPage, isFetchingNextPage]);
 
+  const renderSpinner = () => {
+    if (hasNextPage) return <Spinner size="lg" />;
+    else return null;
+  };
+
   return (
-    <ScrollView
-      w="full"
-      bg={baseBgColor}
-      onScroll={({ nativeEvent }) => {
-        if (hasNextPage && isCloseToBottom(nativeEvent)) {
-          loadNextPosts();
-        }
+    <FlatList
+      ListHeaderComponent={topComponent}
+      data={posts}
+      onEndReached={loadNextPosts}
+      onEndReachedThreshold={0.1}
+      ListFooterComponent={renderSpinner}
+      ItemSeparatorComponent={Divider}
+      bgColor={baseBgColor}
+      keyExtractor={(item, index) => {
```

Comment:
Weird, I didnt get a key component error. Also you should use the id of the post as the key instead of the index

Reply:
If you can find a way to make this change without a key component error, i'll happily abandon this diff lol. Maybe a skissue? but I couldn't get it to not throw that error without adding keys everywhere